spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/service/impl/DefaultTaskJobService.java

```diff
@@ -181,9 +190,19 @@ public void restartJobExecution(long jobExecutionId) throws NoSuchJobExecutionEx
 		TaskExecution taskExecution = this.taskExplorer.getTaskExecution(taskJobExecution.getTaskId());
 		TaskDefinition taskDefinition = this.taskDefinitionRepository.findById(taskExecution.getTaskName())
 				.orElseThrow(() -> new NoSuchTaskDefinitionException(taskExecution.getTaskName()));
-		taskExecutionService.executeTask(taskDefinition.getName(), taskDefinition.getProperties(),
```

Comment:
not sure how passing `taskDefintion.getProperties()` here was valid, maybe we just never caught it.  It should be deploymentProperties, not the task application properties that are passed in.

Reply:
The reason we don't persist the task deployment properties is causing this. Created https://github.com/spring-cloud/spring-cloud-dataflow/issues/2807 to track this.

Reply:
I am asking a different question, passing in task definition properties here should never work, I'm implying that given the validation checks we have on deployment properties to start with a known prefix, relaunching a task such as `timestamp --format='YYYY MM DD'` would not work since `format` does not start with the prefix `app`, `deployer` or `scheduler` (See `DeploymentPropertiesUtils.validateDeploymentProperties`

Reply:
The reason we didn't catch this before is that I moved the deployment property validation from the controller into the service, so while passing this in is wrong, it didn't have any side effect.